json-rpc/src/stream_rpc/logging.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use diem_logger::Schema;
+use serde::Serialize;
+
+#[derive(Clone, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum StreamRpcAction<'a> {
+    ClientConnectionLog(ClientConnectionLog<'a>),
+    HttpRequestLog(HttpRequestLog<'a>),
+}
+
+#[derive(Schema)]
+pub struct StreamRpcLog<'a> {
+    action: StreamRpcAction<'a>,
+}
```

Comment:
hohoho that is one hell of a way to do it.

Reply:
updated :-P